extensions/amp-video-docking/0.1/amp-video-docking.js

```diff
@@ -725,43 +730,46 @@ export class VideoDocking {
       return;
     }
 
-    const currentlyDocked = this.currentlyDocked_;
-
-    if (
-      currentlyDocked &&
-      currentlyDocked.step >= 0 &&
-      layoutRectEquals(currentlyDocked.target.rect, target.rect)
-    ) {
-      return;
-    }
-
-    this.dockInTransferLayerStep_(video, target);
+    this.dockInTransferLayerStep_(
+      video,
+      target,
+      /* step */ 0.1,
+      opt_clientRect
+    );
   }
 
   /**
    * @param {!VideoOrBaseElementDef} video
    * @param {!DockTargetDef} target
-   * @param {number=} opt_step
+   * @param {number} step
+   * @param {!LayoutRectDef=} opt_clientRect
    * @return {!Promise}
    * @private
    */
-  dockInTransferLayerStep_(video, target, opt_step) {
+  dockInTransferLayerStep_(video, target, step, opt_clientRect) {
     // Do this in a multi-step process due to a browser quirk in transferring
     // layers to GPU.
-    const step = opt_step || 0.1;
     // This cutoff is arbitrary and may be dependant on performance.
-    // TODO(alanorozco): Investigate.
```

Comment:
Do you know what all this "step" logic is about?

Reply:
I don't understand it 100%, but it appears it tries to sync animation with scrolling in a limited way. That way the animation is more seamless with scrolling.